// @hexlink/contracts/types/AuthRegistry.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
import {
  BaseContract,
  BigNumber,
  BytesLike,
  CallOverrides,
  ContractTransaction,
  Overrides,
  PopulatedTransaction,
  Signer,
  utils,
} from "ethers";
import { FunctionFragment, Result, EventFragment } from "@ethersproject/abi";
import { Listener, Provider } from "@ethersproject/providers";
import { TypedEventFilter, TypedEvent, TypedListener, OnEvent } from "./common";

export interface AuthRegistryInterface extends utils.Interface {
  contractName: "AuthRegistry";
  functions: {
    "getAuthMetadata(address)": FunctionFragment;
    "setAuthMetadata(string)": FunctionFragment;
  };

  encodeFunctionData(
    functionFragment: "getAuthMetadata",
    values: [string]
  ): string;
  encodeFunctionData(
    functionFragment: "setAuthMetadata",
    values: [string]
  ): string;

  decodeFunctionResult(
    functionFragment: "getAuthMetadata",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "setAuthMetadata",
    data: BytesLike
  ): Result;

  events: {
    "MetadataUpdated(address,bytes32)": EventFragment;
  };

  getEvent(nameOrSignatureOrTopic: "MetadataUpdated"): EventFragment;
}

export type MetadataUpdatedEvent = TypedEvent<
  [string, string],
  { validator: string; hash: string }
>;

export type MetadataUpdatedEventFilter = TypedEventFilter<MetadataUpdatedEvent>;

export interface AuthRegistry extends BaseContract {
  contractName: "AuthRegistry";
  connect(signerOrProvider: Signer | Provider | string): this;
  attach(addressOrName: string): this;
  deployed(): Promise<this>;

  interface: AuthRegistryInterface;

  queryFilter<TEvent extends TypedEvent>(
    event: TypedEventFilter<TEvent>,
    fromBlockOrBlockhash?: string | number | undefined,
    toBlock?: string | number | undefined
  ): Promise<Array<TEvent>>;

  listeners<TEvent extends TypedEvent>(
    eventFilter?: TypedEventFilter<TEvent>
  ): Array<TypedListener<TEvent>>;
  listeners(eventName?: string): Array<Listener>;
  removeAllListeners<TEvent extends TypedEvent>(
    eventFilter: TypedEventFilter<TEvent>
  ): this;
  removeAllListeners(eventName?: string): this;
  off: OnEvent<this>;
  on: OnEvent<this>;
  once: OnEvent<this>;
  removeListener: OnEvent<this>;

  functions: {
    getAuthMetadata(
      owner: string,
      overrides?: CallOverrides
    ): Promise<[string]>;

    setAuthMetadata(
      metadata: string,
      overrides?: Overrides & { from?: string | Promise<string> }
    ): Promise<ContractTransaction>;
  };

  getAuthMetadata(owner: string, overrides?: CallOverrides): Promise<string>;

  setAuthMetadata(
    metadata: string,
    overrides?: Overrides & { from?: string | Promise<string> }
  ): Promise<ContractTransaction>;

  callStatic: {
    getAuthMetadata(owner: string, overrides?: CallOverrides): Promise<string>;

    setAuthMetadata(metadata: string, overrides?: CallOverrides): Promise<void>;
  };

  filters: {
    "MetadataUpdated(address,bytes32)"(
      validator?: string | null,
      hash?: BytesLike | null
    ): MetadataUpdatedEventFilter;
    MetadataUpdated(
      validator?: string | null,
      hash?: BytesLike | null
    ): MetadataUpdatedEventFilter;
  };

  estimateGas: {
    getAuthMetadata(
      owner: string,
      overrides?: CallOverrides
    ): Promise<BigNumber>;

    setAuthMetadata(
      metadata: string,
      overrides?: Overrides & { from?: string | Promise<string> }
    ): Promise<BigNumber>;
  };

  populateTransaction: {
    getAuthMetadata(
      owner: string,
      overrides?: CallOverrides
    ): Promise<PopulatedTransaction>;

    setAuthMetadata(
      metadata: string,
      overrides?: Overrides & { from?: string | Promise<string> }
    ): Promise<PopulatedTransaction>;
  };
}
