// @hexlink/contracts/types/IERC4972Registry.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
import {
  BaseContract,
  BigNumber,
  BytesLike,
  CallOverrides,
  PopulatedTransaction,
  Signer,
  utils,
} from "ethers";
import { FunctionFragment, Result } from "@ethersproject/abi";
import { Listener, Provider } from "@ethersproject/providers";
import { TypedEventFilter, TypedEvent, TypedListener, OnEvent } from "./common";

export interface IERC4972RegistryInterface extends utils.Interface {
  contractName: "IERC4972Registry";
  functions: {
    "getNameService()": FunctionFragment;
    "getOwnedAccount(bytes32)": FunctionFragment;
  };

  encodeFunctionData(
    functionFragment: "getNameService",
    values?: undefined
  ): string;
  encodeFunctionData(
    functionFragment: "getOwnedAccount",
    values: [BytesLike]
  ): string;

  decodeFunctionResult(
    functionFragment: "getNameService",
    data: BytesLike
  ): Result;
  decodeFunctionResult(
    functionFragment: "getOwnedAccount",
    data: BytesLike
  ): Result;

  events: {};
}

export interface IERC4972Registry extends BaseContract {
  contractName: "IERC4972Registry";
  connect(signerOrProvider: Signer | Provider | string): this;
  attach(addressOrName: string): this;
  deployed(): Promise<this>;

  interface: IERC4972RegistryInterface;

  queryFilter<TEvent extends TypedEvent>(
    event: TypedEventFilter<TEvent>,
    fromBlockOrBlockhash?: string | number | undefined,
    toBlock?: string | number | undefined
  ): Promise<Array<TEvent>>;

  listeners<TEvent extends TypedEvent>(
    eventFilter?: TypedEventFilter<TEvent>
  ): Array<TypedListener<TEvent>>;
  listeners(eventName?: string): Array<Listener>;
  removeAllListeners<TEvent extends TypedEvent>(
    eventFilter: TypedEventFilter<TEvent>
  ): this;
  removeAllListeners(eventName?: string): this;
  off: OnEvent<this>;
  on: OnEvent<this>;
  once: OnEvent<this>;
  removeListener: OnEvent<this>;

  functions: {
    getNameService(overrides?: CallOverrides): Promise<[string]>;

    getOwnedAccount(
      name: BytesLike,
      overrides?: CallOverrides
    ): Promise<[string]>;
  };

  getNameService(overrides?: CallOverrides): Promise<string>;

  getOwnedAccount(name: BytesLike, overrides?: CallOverrides): Promise<string>;

  callStatic: {
    getNameService(overrides?: CallOverrides): Promise<string>;

    getOwnedAccount(
      name: BytesLike,
      overrides?: CallOverrides
    ): Promise<string>;
  };

  filters: {};

  estimateGas: {
    getNameService(overrides?: CallOverrides): Promise<BigNumber>;

    getOwnedAccount(
      name: BytesLike,
      overrides?: CallOverrides
    ): Promise<BigNumber>;
  };

  populateTransaction: {
    getNameService(overrides?: CallOverrides): Promise<PopulatedTransaction>;

    getOwnedAccount(
      name: BytesLike,
      overrides?: CallOverrides
    ): Promise<PopulatedTransaction>;
  };
}
