// @hexlink/contracts/types/IERC6662.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
import {
  BaseContract,
  BigNumber,
  BytesLike,
  CallOverrides,
  PopulatedTransaction,
  Signer,
  utils,
} from "ethers";
import { FunctionFragment, Result } from "@ethersproject/abi";
import { Listener, Provider } from "@ethersproject/providers";
import { TypedEventFilter, TypedEvent, TypedListener, OnEvent } from "./common";

export interface IERC6662Interface extends utils.Interface {
  contractName: "IERC6662";
  functions: {
    "getAuthRegistry()": FunctionFragment;
  };

  encodeFunctionData(
    functionFragment: "getAuthRegistry",
    values?: undefined
  ): string;

  decodeFunctionResult(
    functionFragment: "getAuthRegistry",
    data: BytesLike
  ): Result;

  events: {};
}

export interface IERC6662 extends BaseContract {
  contractName: "IERC6662";
  connect(signerOrProvider: Signer | Provider | string): this;
  attach(addressOrName: string): this;
  deployed(): Promise<this>;

  interface: IERC6662Interface;

  queryFilter<TEvent extends TypedEvent>(
    event: TypedEventFilter<TEvent>,
    fromBlockOrBlockhash?: string | number | undefined,
    toBlock?: string | number | undefined
  ): Promise<Array<TEvent>>;

  listeners<TEvent extends TypedEvent>(
    eventFilter?: TypedEventFilter<TEvent>
  ): Array<TypedListener<TEvent>>;
  listeners(eventName?: string): Array<Listener>;
  removeAllListeners<TEvent extends TypedEvent>(
    eventFilter: TypedEventFilter<TEvent>
  ): this;
  removeAllListeners(eventName?: string): this;
  off: OnEvent<this>;
  on: OnEvent<this>;
  once: OnEvent<this>;
  removeListener: OnEvent<this>;

  functions: {
    getAuthRegistry(overrides?: CallOverrides): Promise<[string]>;
  };

  getAuthRegistry(overrides?: CallOverrides): Promise<string>;

  callStatic: {
    getAuthRegistry(overrides?: CallOverrides): Promise<string>;
  };

  filters: {};

  estimateGas: {
    getAuthRegistry(overrides?: CallOverrides): Promise<BigNumber>;
  };

  populateTransaction: {
    getAuthRegistry(overrides?: CallOverrides): Promise<PopulatedTransaction>;
  };
}
