// @hexlink/contracts/dist/types/factories/ERC4972Account__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ERC4972Account__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "owner",
                type: "address",
            },
        ],
        name: "InvalidNameOwner",
        type: "error",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "name",
                type: "bytes32",
            },
        ],
        name: "InvalidNameToSet",
        type: "error",
    },
    {
        inputs: [],
        name: "OnlySelfCallAllowed",
        type: "error",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        name: "NameOwnerUpdated",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "bytes32",
                name: "name",
                type: "bytes32",
            },
        ],
        name: "NameUpdated",
        type: "event",
    },
    {
        inputs: [],
        name: "getERC4972Registry",
        outputs: [
            {
                internalType: "contract IERC4972Registry",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [],
        name: "getName",
        outputs: [
            {
                internalType: "bytes32",
                name: "",
                type: "bytes32",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [],
        name: "getNameOwner",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "name",
                type: "bytes32",
            },
        ],
        name: "setName",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [],
        name: "setNameOwner",
        outputs: [
            {
                internalType: "bool",
                name: "updated",
                type: "bool",
            },
        ],
        stateMutability: "nonpayable",
        type: "function",
    },
];
var ERC4972Account__factory = exports.ERC4972Account__factory = /** @class */ (function () {
    function ERC4972Account__factory() {
    }
    ERC4972Account__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    ERC4972Account__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    ERC4972Account__factory.abi = _abi;
    return ERC4972Account__factory;
}());
