// @hexlink/contracts/dist/types/factories/IAuthRegistry__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IAuthRegistry__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [
            {
                internalType: "address",
                name: "validator",
                type: "address",
            },
        ],
        name: "getAuthMetadata",
        outputs: [
            {
                internalType: "string",
                name: "",
                type: "string",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IAuthRegistry__factory = exports.IAuthRegistry__factory = /** @class */ (function () {
    function IAuthRegistry__factory() {
    }
    IAuthRegistry__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IAuthRegistry__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IAuthRegistry__factory.abi = _abi;
    return IAuthRegistry__factory;
}());
