// @hexlink/contracts/dist/types/factories/IERC4972Registry__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IERC4972Registry__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [],
        name: "getNameService",
        outputs: [
            {
                internalType: "contract INameService",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "bytes32",
                name: "name",
                type: "bytes32",
            },
        ],
        name: "getOwnedAccount",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IERC4972Registry__factory = exports.IERC4972Registry__factory = /** @class */ (function () {
    function IERC4972Registry__factory() {
    }
    IERC4972Registry__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IERC4972Registry__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IERC4972Registry__factory.abi = _abi;
    return IERC4972Registry__factory;
}());
