// @hexlink/contracts/dist/types/factories/IERC6662__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IERC6662__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [],
        name: "getAuthRegistry",
        outputs: [
            {
                internalType: "contract IAuthRegistry",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IERC6662__factory = exports.IERC6662__factory = /** @class */ (function () {
    function IERC6662__factory() {
    }
    IERC6662__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IERC6662__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IERC6662__factory.abi = _abi;
    return IERC6662__factory;
}());
