// @hexlink/contracts/dist/types/factories/IERC4972Account__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.IERC4972Account__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [],
        name: "getERC4972Registry",
        outputs: [
            {
                internalType: "contract IERC4972Registry",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [],
        name: "getName",
        outputs: [
            {
                internalType: "bytes32",
                name: "",
                type: "bytes32",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
];
var IERC4972Account__factory = exports.IERC4972Account__factory = /** @class */ (function () {
    function IERC4972Account__factory() {
    }
    IERC4972Account__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    IERC4972Account__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    IERC4972Account__factory.abi = _abi;
    return IERC4972Account__factory;
}());
