// @hexlink/contracts/dist/types/factories/Simple2FA__factory.js
"use strict";
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Simple2FA__factory = void 0;
var ethers_1 = require("ethers");
var _abi = [
    {
        inputs: [],
        name: "OnlySelfCallAllowed",
        type: "error",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        name: "SecondFactorAdded",
        type: "event",
    },
    {
        anonymous: false,
        inputs: [
            {
                indexed: true,
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        name: "SecondFactorRemoved",
        type: "event",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "factor",
                type: "address",
            },
        ],
        name: "addSecondFactor",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
    {
        inputs: [],
        name: "getSecondFactor",
        outputs: [
            {
                internalType: "address",
                name: "",
                type: "address",
            },
        ],
        stateMutability: "view",
        type: "function",
    },
    {
        inputs: [
            {
                internalType: "address",
                name: "factor",
                type: "address",
            },
        ],
        name: "removeSecondFactor",
        outputs: [],
        stateMutability: "nonpayable",
        type: "function",
    },
];
var Simple2FA__factory = exports.Simple2FA__factory = /** @class */ (function () {
    function Simple2FA__factory() {
    }
    Simple2FA__factory.createInterface = function () {
        return new ethers_1.utils.Interface(_abi);
    };
    Simple2FA__factory.connect = function (address, signerOrProvider) {
        return new ethers_1.Contract(address, _abi, signerOrProvider);
    };
    Simple2FA__factory.abi = _abi;
    return Simple2FA__factory;
}());
