// @hexlink/contracts/types/factories/IAccountInitializer__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import { Provider } from "@ethersproject/providers";
import type {
  IAccountInitializer,
  IAccountInitializerInterface,
} from "../IAccountInitializer";

const _abi = [
  {
    inputs: [
      {
        internalType: "bytes32",
        name: "name",
        type: "bytes32",
      },
      {
        internalType: "address",
        name: "owner",
        type: "address",
      },
    ],
    name: "initialize",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
];

export class IAccountInitializer__factory {
  static readonly abi = _abi;
  static createInterface(): IAccountInitializerInterface {
    return new utils.Interface(_abi) as IAccountInitializerInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): IAccountInitializer {
    return new Contract(address, _abi, signerOrProvider) as IAccountInitializer;
  }
}
