// @hexlink/contracts/types/factories/IERC4972Account__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import { Provider } from "@ethersproject/providers";
import type {
  IERC4972Account,
  IERC4972AccountInterface,
} from "../IERC4972Account";

const _abi = [
  {
    inputs: [],
    name: "getERC4972Registry",
    outputs: [
      {
        internalType: "contract IERC4972Registry",
        name: "",
        type: "address",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
  {
    inputs: [],
    name: "getName",
    outputs: [
      {
        internalType: "bytes32",
        name: "",
        type: "bytes32",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
];

export class IERC4972Account__factory {
  static readonly abi = _abi;
  static createInterface(): IERC4972AccountInterface {
    return new utils.Interface(_abi) as IERC4972AccountInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): IERC4972Account {
    return new Contract(address, _abi, signerOrProvider) as IERC4972Account;
  }
}
