// @hexlink/contracts/types/factories/IERC4972Registry__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import { Provider } from "@ethersproject/providers";
import type {
  IERC4972Registry,
  IERC4972RegistryInterface,
} from "../IERC4972Registry";

const _abi = [
  {
    inputs: [],
    name: "getNameService",
    outputs: [
      {
        internalType: "contract INameService",
        name: "",
        type: "address",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "bytes32",
        name: "name",
        type: "bytes32",
      },
    ],
    name: "getOwnedAccount",
    outputs: [
      {
        internalType: "address",
        name: "",
        type: "address",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
];

export class IERC4972Registry__factory {
  static readonly abi = _abi;
  static createInterface(): IERC4972RegistryInterface {
    return new utils.Interface(_abi) as IERC4972RegistryInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): IERC4972Registry {
    return new Contract(address, _abi, signerOrProvider) as IERC4972Registry;
  }
}
