// @hexlink/contracts/types/factories/INameService__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import { Provider } from "@ethersproject/providers";
import type { INameService, INameServiceInterface } from "../INameService";

const _abi = [
  {
    inputs: [],
    name: "defaultOwner",
    outputs: [
      {
        internalType: "address",
        name: "",
        type: "address",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "bytes32",
        name: "name",
        type: "bytes32",
      },
    ],
    name: "owner",
    outputs: [
      {
        internalType: "address",
        name: "",
        type: "address",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
];

export class INameService__factory {
  static readonly abi = _abi;
  static createInterface(): INameServiceInterface {
    return new utils.Interface(_abi) as INameServiceInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): INameService {
    return new Contract(address, _abi, signerOrProvider) as INameService;
  }
}
