// @hexlink/contracts/types/factories/Simple2FA__factory.ts
/* Autogenerated file. Do not edit manually. */
/* tslint:disable */
/* eslint-disable */

import { Contract, Signer, utils } from "ethers";
import { Provider } from "@ethersproject/providers";
import type { Simple2FA, Simple2FAInterface } from "../Simple2FA";

const _abi = [
  {
    inputs: [],
    name: "OnlySelfCallAllowed",
    type: "error",
  },
  {
    anonymous: false,
    inputs: [
      {
        indexed: true,
        internalType: "address",
        name: "",
        type: "address",
      },
    ],
    name: "SecondFactorAdded",
    type: "event",
  },
  {
    anonymous: false,
    inputs: [
      {
        indexed: true,
        internalType: "address",
        name: "",
        type: "address",
      },
    ],
    name: "SecondFactorRemoved",
    type: "event",
  },
  {
    inputs: [
      {
        internalType: "address",
        name: "factor",
        type: "address",
      },
    ],
    name: "addSecondFactor",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
  {
    inputs: [],
    name: "getSecondFactor",
    outputs: [
      {
        internalType: "address",
        name: "",
        type: "address",
      },
    ],
    stateMutability: "view",
    type: "function",
  },
  {
    inputs: [
      {
        internalType: "address",
        name: "factor",
        type: "address",
      },
    ],
    name: "removeSecondFactor",
    outputs: [],
    stateMutability: "nonpayable",
    type: "function",
  },
];

export class Simple2FA__factory {
  static readonly abi = _abi;
  static createInterface(): Simple2FAInterface {
    return new utils.Interface(_abi) as Simple2FAInterface;
  }
  static connect(
    address: string,
    signerOrProvider: Signer | Provider
  ): Simple2FA {
    return new Contract(address, _abi, signerOrProvider) as Simple2FA;
  }
}
